pkg/controllermanager/controller/shoot/shoot_maintenance_test.go

```diff
@@ -196,22 +218,46 @@ var _ = Describe("Shoot Maintenance", func() {
 		It("should return an error - cloud profile has no matching (machineImage.name) machine image defined", func() {
 			cloudProfile.Spec.MachineImages = cloudProfile.Spec.MachineImages[1:]
 
-			_, err := MaintainMachineImages(shoot, cloudProfile, machineCurrentImages)
+			_, _, err := MaintainMachineImages(testlogger, shoot, cloudProfile)
 
 			Expect(err).NotTo(BeNil())
 		})
+
+		It("should return an error and not downgrade shoot - even if highest version is expired", func() {
```

Comment:
Nice, can you also do that for the case, that the shoot uses a preview version?

Reply:
Btw, what is the desired behaviour, if the Shoot uses a preview version and there is a higher supported version? (Not sure, if that would be even a valid configuration of a CloudProfile - I'm losing track of all the different scenarios and rules 🙈).
Will it then be upgraded to the higher supported version?

Reply:
Ok, will you take care of that?

Reply:
test case for [the logic is here](https://github.com/gardener/gardener/blob/2e5c630352e0e5dc5db661b309a51b705e550fc1/pkg/controllermanager/controller/shoot/shoot_maintenance_test.go#L240).

Reply:
See above.